server/dump.go

```diff
@@ -90,3 +98,52 @@ func (p *Plugin) createDump(ctx context.Context, min, max time.Time, remoteStora
 
 	return zipFileNameRemote, nil
 }
+
+// Metadata is the auxiliary data added to a plugin downloadable.
+// It should contain the required fields and may have it's custom values as well.
+// TODO: move this to mattermost/server/public/pluginapi in the future
+type metadata struct {
+	ServerVersion string         `yaml:"server_version"`
+	ServerID      string         `yaml:"server_id"`
+	LicenseID     string         `yaml:"license_id"`
+	PluginID      string         `yaml:"plugin_id"`
+	Custom        map[string]any `yaml:"custom"`
+}
+
+func (p *Plugin) genarateMetadataForDump(path string, min, max time.Time) (*metadata, error) {
+	m := &metadata{}
```

Comment:
open q: Is there any (aggregated) value, metric or metadata piece useful/valuable specific from the Prometheus package metadata that needs to/could be elevated into this more general metadata package (and allows us to avoid opening the Prometheus section of the package but still maybe report on it)?

Reply:
Good point, as some sort of brainstorming; we can run [promQL](https://prometheus.io/docs/prometheus/latest/querying/basics/) on the TSDB very efficiently, which means we can actually report the metrics that are off from what we expect.